test/test-mc-fle2-encryption-placeholder.c

```diff
@@ -0,0 +1,448 @@
+                                         "800000000000000694010707265636973696f6e00020000000012636d000000000000"
+                                         "000000127300010000000000000000");
+        ASSERT(bson_init_static(&as_bson, buf.data + 1, buf.len - 1));
+        mc_FLE2EncryptionPlaceholder_init(&placeholder);
+        ASSERT_OK_STATUS(mc_FLE2EncryptionPlaceholder_parse(&placeholder, &as_bson, status), status);
+
+        ASSERT(placeholder.type == MONGOCRYPT_FLE2_PLACEHOLDER_TYPE_INSERT);
+        ASSERT(placeholder.algorithm == MONGOCRYPT_FLE2_ALGORITHM_RANGE);
+
+        _mongocrypt_buffer_t expect_index_key_id;
+        _mongocrypt_buffer_copy_from_hex(&expect_index_key_id, "12345678123498761234123456789012");
+        ASSERT_CMPBUF(placeholder.index_key_id, expect_index_key_id);
+        _mongocrypt_buffer_cleanup(&expect_index_key_id);
+
+        _mongocrypt_buffer_t expect_user_key_id;
+        _mongocrypt_buffer_copy_from_hex(&expect_user_key_id, "abcdefab123498761234123456789012");
+        ASSERT_CMPBUF(placeholder.user_key_id, expect_user_key_id);
+        _mongocrypt_buffer_cleanup(&expect_user_key_id);
+
+        ASSERT_CMPINT64(placeholder.sparsity, ==, 1);
+
+        // Parse FLE2RangeInsertSpec.
+        {
+            mc_FLE2RangeInsertSpec_t spec;
+
+            ASSERT_OK_STATUS(mc_FLE2RangeInsertSpec_parse(&spec, &placeholder.v_iter, false /* use_range_v2 */, status),
+                             status);
+
+            ASSERT(BSON_ITER_HOLDS_DOUBLE(&spec.v));
+            ASSERT_CMPDOUBLE(bson_iter_double(&spec.v), ==, 123.456);
+
+            ASSERT(BSON_ITER_HOLDS_DOUBLE(&spec.min));
+            ASSERT_CMPDOUBLE(bson_iter_double(&spec.min), ==, 0.0);
+
+            ASSERT(BSON_ITER_HOLDS_DOUBLE(&spec.max));
+            ASSERT_CMPDOUBLE(bson_iter_double(&spec.max), ==, 200.0);
+
+            ASSERT(spec.precision.set);
+            ASSERT_CMPUINT32(spec.precision.value, ==, 2);
+        }
+
+        mc_FLE2EncryptionPlaceholder_cleanup(&placeholder);
+        _mongocrypt_buffer_cleanup(&buf);
+        mongocrypt_status_destroy(status);
+    }
+}
+
+static bool _parse_text_search_spec_from_placeholder(_mongocrypt_tester_t *tester,
+                                                     const char *spec_json_in,
+                                                     mc_FLE2TextSearchInsertSpec_t *spec_out,
+                                                     mongocrypt_status_t *status_out) {
+    const char *template = RAW_STRING({
+        "t" : {"$numberInt" : "1"},
+        "a" : {"$numberInt" : "4"},
+        "ki" : {"$binary" : {"base64" : "EjRWeBI0mHYSNBI0VniQEg==", "subType" : "04"}},
+        "ku" : {"$binary" : {"base64" : "q83vqxI0mHYSNBI0VniQEg==", "subType" : "04"}},
+        "v" : % s,
+        "cm" : {"$numberLong" : "7"}
+    });
+    bson_t *const as_bson = TMP_BSON(template, spec_json_in);
+
+    mongocrypt_status_t *inner_status = mongocrypt_status_new();
```

Comment:
Is there a reason we have an inner status here, rather than using the passed status for everything? If we get a non-OK status, we fail anyways, so it shouldn't matter, right?

Reply:
yep. removed